api/src/main/java/ai/djl/nn/AbstractBlock.java

```diff
@@ -92,19 +94,41 @@ public AbstractBlock(byte version) {
     /**
      * Use this to add a child block to this block.
      *
-     * @param name Name of the block, must be unique or otherwise existing children with this name
-     *     are removed, must not be null.
+     * @param name Name of the block, must not be null.
      * @param block The block, must not be null.
      * @param <B> The type of block
      * @return the block given as a parameter - that way the block can be created and reassigned to
      *     a member variable more easily.
      */
-    protected final <B extends Block> B addChildBlock(String name, B block) {
+    public final <B extends Block> B addChildBlock(String name, B block) {
```

Comment:
Is there a reason this is being made public? It is intended only to be used within the block when constructing it.

Reply:
I made it public as in my own scenario, I want to call `addChildBlock `on a `SequentialBlock`.
`SequentialBlock` has itself a few `add()` methods, but you cannot define a name for the block to be added, that's a pitty. 

As general difference, `addChildBlock ` would allow to define a name as well and return the child block, while `add ` doesn't allow to define the name (a generic name is used) and returns the parent block itself instead of the child block.

Reply:
Maybe its not a good idea to have two different API styles, `add `and `addChildBlock`, in parallel and keep `addChildBlock ` protected. If 
https://github.com/deepjavalibrary/djl/pull/1726 can be extended to support a name for any type of block, then I can adjust this PR to it.

Reply:
The main reason is that addChildBlock was intended to be internal because for most kinds of blocks, it should only be called internally. The add was external from SequentialBlock because SequentialBlock is like a "block-builder" and it should actually have a public add. If you want to create more public adds or add with names, they should be on sequentialBlock rather than the general block